File: rapp-examples/rapp-test1/test1.cpp
/* 
 * We assume that STL is present on the robot - 
 * ANY other header/library must be imported explicitly in the RAPP Submission Service
 */
#include <iostream>

// RAPP API
#include "rapp.hpp"

// this is just a mock-up paradigm of a RAPP written in pure C++
int main ( void )
{
    std::cout << rapp::services::FooSync( "foo" ) << std::endl;
    std::cout << rapp::services::FooAsync( "foo", []( void ){ std::cout << "async cb" << std::endl;} ) << std::endl;
    std::cout << rapp::core::Foo( "foo" ) << std::endl;
}

File: rapp-examples/rapp-test1/test1.js
#!/usr/bin/env node

/**
 * This is a mock-up of a Javascript RAPP.
 * It relies on Node.js for importing other modules, such as the RAPP API.
 * 
 * NOTE: We will NOT ALLOW devs, to mix and match Javascript and C++, only one language is to be used for a RAPP.
 */

// Import the RAPP JS API
var rapp = require('./rapp.js');

//console.log ( rapp );

var services = new rapp.RAPPServices( "alex", "qwepoi" );

console.log( services.FooSync( "foo" ) );
console.log( services.FooAsync( "foo" ) );

var core = new rapp.RAPPCore();

console.log( core.Foo( "foo" ) );